lib/ansible/module_utils/service.py

```diff
@@ -207,17 +207,19 @@ def daemonize(module, cmd):
         p = subprocess.Popen(run_cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, preexec_fn=lambda: os.close(pipe[1]))
         fds = [p.stdout, p.stderr]
 
-        # loop reading output till its done
+        # loop reading output till it is done
         output = {p.stdout: b(""), p.stderr: b("")}
         while fds:
             rfd, wfd, efd = select.select(fds, [], fds, 1)
-            if (rfd + wfd + efd) or p.poll():
+            if (rfd + wfd + efd) or p.poll() is None:
```

Comment:
the 'is None' changes the meaning as the existing condition would trigger either if we have output OR the process ended 'is not None, but returns an rc that is not 0', if changing this way i would suggest changing the `or` to an `and`.

Reply:
`is not None` seems to better reflect the intent ... only try to read if either the descriptors are ready or the program terminated (this time including success with 0, not just errors with 1+)

Reply:
@bcoca, @webknjaz no, the original condition is incorrect: that was tested. The logic here is to read if either the descriptors are ready or the program *has not* terminated.